packages/@headlessui-react/src/components/listbox/listbox.tsx

```diff
@@ -1249,40 +851,50 @@ function OptionFn<
     }
   })
 
+  let [listboxState, activationTrigger, __demoMode, activeOptionIndex] = useSlice(
+    machine,
+    (state) => [
+      state.listboxState,
+      state.activationTrigger,
+      state.__demoMode,
+      state.activeOptionIndex,
+    ]
+  )
   useIsoMorphicEffect(() => {
-    if (data.__demoMode) return
-    if (data.listboxState !== ListboxStates.Open) return
+    if (__demoMode) return
+    if (listboxState !== ListboxStates.Open) return
     if (!active) return
-    if (data.activationTrigger === ActivationTrigger.Pointer) return
+    if (activationTrigger === ActivationTrigger.Pointer) return
     return disposables().requestAnimationFrame(() => {
       internalOptionRef.current?.scrollIntoView?.({ block: 'nearest' })
     })
   }, [
     internalOptionRef,
     active,
-    data.__demoMode,
-    data.listboxState,
-    data.activationTrigger,
-    /* We also want to trigger this when the position of the active item changes so that we can re-trigger the scrollIntoView */ data.activeOptionIndex,
+    __demoMode,
+    listboxState,
+    activationTrigger,
+    /* We also want to trigger this when the position of the active item changes so that we can re-trigger the scrollIntoView */
+    activeOptionIndex,
```

Comment:
Should this be abstracted into a useSlice for shouldScrollIntoView like we did with Menu or nah?

Reply:
Yep, that happens in a later commit: https://github.com/tailwindlabs/headlessui/pull/3688/commits/4ee5b37c971ec4b36f31936fcf1835a2e1c4db08